src/main/kotlin/me/serce/solidity/ide/formatting/SolidityFormattingRestriction.kt

```diff
@@ -0,0 +1,14 @@
+package me.serce.solidity.ide.formatting
+
+import com.intellij.lang.LanguageFormattingRestriction
+import com.intellij.psi.PsiElement
+import me.serce.solidity.settings.FormatterType
+import me.serce.solidity.settings.SoliditySettings
+
+class SolidityFormattingRestriction : LanguageFormattingRestriction {
```

Comment:
The restriction also applies globally, so disabling the formatter disables it globally for all files, let me merge the PR and I'll fix it in the follow-up.